src/Files.App/Data/Models/StatusBanner.cs

```diff
@@ -0,0 +1,196 @@
+﻿using Microsoft.UI.Xaml.Controls;
+using System.Windows.Input;
+
+namespace Files.App.Data.Models
+{
+	public class StatusBanner : ObservableObject
+	{
+		private readonly float initialProgress = 0.0f;
+
+		private string fullTitle;
+
+		private bool isCancelled;
+
+		private int progress = 0;
+		public int Progress
+		{
+			get => progress;
+			set => SetProperty(ref progress, value);
+		}
+
+		private bool isProgressing = false;
+
+		public bool IsProgressing
+		{
+			get => isProgressing;
+			set => SetProperty(ref isProgressing, value);
+		}
+
+		public string Title { get; private set; }
+
+		private ReturnResult status = ReturnResult.InProgress;
+		public ReturnResult Status
+		{
+			get => status;
+			set => SetProperty(ref status, value);
+		}
+
+		public FileOperationType Operation { get; private set; }
+
+		public string Message { get; private set; }
+
+		public InfoBarSeverity InfoBarSeverity { get; private set; } = InfoBarSeverity.Informational;
+
+		public string PrimaryButtonText { get; set; }
+
+		public string SecondaryButtonText { get; set; } = "Cancel";
+
+		public Action PrimaryButtonClick { get; }
+
+		public ICommand CancelCommand { get; }
+
+		public bool SolutionButtonsVisible { get; } = false;
+
+		public bool CancelButtonVisible
+			=> CancellationTokenSource is not null;
+
+		public CancellationTokenSource CancellationTokenSource { get; set; }
+
+		public string FullTitle
+		{
+			get => fullTitle;
+			set => SetProperty(ref fullTitle, value ?? string.Empty);
+		}
+
+		public bool IsCancelled
+		{
+			get => isCancelled;
+			set => SetProperty(ref isCancelled, value);
+		}
```

Comment:
These two are the only ones separated from their backing fields

Reply:
Fixed